### ISSUE TITLE
html tables with template tags (but not inside td tag)

### ISSUE THREAD

**Issue by joegsn** [wkhtmltopdf/wkhtmltopdf#2753]
It seems like wkhtmltopdf does not like <template> tags inside of a <table> or <tr> tag.  The <template> tag was introduced as part of html5, and intentionally is supposed to be supported as a child of nearly any other element, including <table> and <tr>.

I've been utilizing them in an experiment with vue.js, so I know it works in the major web browsers.  I went to try it with wkhtmltopdf, and received unexpected results.  It moved the <template> tag outside of the <table>.

Here's the test.html code I have for demonstration purposes:
```html
<!DOCTYPE html>
<html>
 <head>
  <script src='http://cdnjs.cloudflare.com/ajax/libs/vue/1.0.14/vue.min.js' type="text/javascript"></script>
 </head>
 <body>
  <div id='page'>
   <div id='content'>
    <h1 class='ui-widget-header ui-corner-top'>Job List</h1>
    <table id="job_report" class="job_report">
     <thead>
      <tr>
       <th class='nowrap'> Job # </th>
      </tr>
     </thead>
     <tbody>
      <template v-for='entry in report.jobs'>
       <tr>
        <td colspan=4>{{entry.number}}</td>
       </tr>
      </template>
     </tbody>
    </table>
    <script type='text/javascript'>
     lastnote = null;
     jobdata = new Vue({
      el: '#job_report',
      data: {
       report: {'jobs': [{"number":"12345"}, {"number":"67890"}]},
      },
     });
    </script>
   </div>
  </div>
 </body>
</html>
```

I've been running wkhtmltopdf using the following command:
`wkhtmltopdf --debug-javascript test.html test.pdf`

Executing this will give you a PDF that does not have any table data filled in (unlike if you were to view the html data in a web browser with the exception of IE).  The modern web browsers (chrome, firefox, opera, safari, webkit, edge) support the template tag within a table just fine.  I believe wkhtmltopdf is moving the template tag outside of the table (just like if it were a div tag).

**Comment by zenmodbr** [comment]
Hi joegsn. Did you find a solution for this ? I'm having the exactly same issue.
Thx!

**Comment by PhilterPaper** [comment]
joegsn, please edit your post and put backticks around HTML tags in your text, That way they'll show up, and we can see what you're talking about. "Preview" your posts so you can catch stuff disappearing like this.

**Comment by joegsn** [comment]
@zenmodbr No, I have not found a solution for it, though I did find a (temporary?) workaround for it (more below).  I believe the problem ultimately lies in the QtWebkit code, as I had the exact same issue trying to use PhantomJS, which is another QtWebkit-based product.  I'm not sure if newer versions of QtWebkit have fixed the issue, and I've been largely migrating away from wkhtmltopdf in favor of generating the PDF content itself through libraries for the languages I'm using.

The workaround I employed for a while was to use SlimerJS to pre-process the webpage.  It relies on the Firefox rendering engine, so it has a different operational ability than the QtWebkit-based programs (like wkhtmltopdf & phantomjs).  It meant that I was taking the page I wanted to render, passing it to slimerjs, then passing the result to wkhtmltopdf.  It was more steps, and a little fragile, but it worked until I eventually replaced the code with direct PDF generation (no html intermediary).

@PhilterPaper Thanks, I had not realized that the tags even disappeared.  Might explain why no one else commented for a year.
